views/js/runner/plugins/content/dialog/dialog.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2016 (original work) Open Assessment Technologies SA ;
+ */
+
+/**
+ * Test Runner Content Plugin : Overlay
+ *
+ * @author Bertrand Chevrier <bertrand@taotesting.com>
+ */
+define([
+    'jquery',
+    'i18n',
+    'taoTests/runner/plugin',
+    'ui/dialog/alert',
+    'ui/dialog/confirm',
+], function ($, __, pluginFactory, dialogAlert, dialogConfirm){
+    'use strict';
+
+    /**
+     * Returns the configured plugin
+     */
+    return pluginFactory({
+        name : 'dialog',
+
+        /**
+         * Initialize the plugin (called during runner's init)
+         */
+        init : function init(){
+            var self = this;
+            var testRunner = this.getTestRunner();
+
+            //change plugin state
+            testRunner
+                .on('alert',  dialogAlert)
+                .on('confirm', dialogConfirm);
```

Comment:
I have a concern about this: how can we be aware of the user's response ? Is it by providing the callback through the event triggering ? In that case, the 'alert' and 'confirm' events must always be listened by one plugin, otherwise the callback will never be called and the runner can be locked.